iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -0,0 +1,225 @@
+        {
+            while (true)
+            {
+                _logger.LogInformation($"Device waiting for C2D messages from the hub - for {s_sleepDuration}...");
+                _logger.LogInformation("Use the IoT Hub Azure Portal or Azure IoT Explorer to send a message to this device.");
+
+                try
+                {
+                    await ReceiveMessageAndCompleteAsync();
+                }
+                catch (Exception ex) when (ex is DeviceMessageLockLostException)
+                {
+                    _logger.LogWarning($"Attempted to complete a received message whose lock token has expired; ignoring: {ex}");
+                }
+                catch (Exception ex) when (ex is IotHubException exception && exception.IsTransient)
+                {
+                    // Inspect the exception to figure out if operation should be retried, or if user-input is required.
+                    _logger.LogError($"An IotHubException was caught, but will try to recover and retry explicitly: {ex}");
+                }
+                catch (Exception ex) when (ExceptionHelper.IsNetworkExceptionChain(ex))
+                {
+                    _logger.LogError($"A network related exception was caught, but will try to recover and retry explicitly: {ex}");
+                }
+            }
+        }
+
+        private async Task SendMessageAsync(int count)
+        {
+            var _temperature = s_randomGenerator.Next(20, 35);
+            var _humidity = s_randomGenerator.Next(60, 80);
+            string dataBuffer = $"{{\"temperature\":{_temperature},\"humidity\":{_humidity}}}";
+
+            using var eventMessage = new Message(Encoding.UTF8.GetBytes(dataBuffer))
+            {
+                MessageId = count.ToString(),
+                ContentEncoding = Encoding.UTF8.ToString(),
+                ContentType = "application/json",
+            };
+            eventMessage.Properties.Add("temperatureAlert", (_temperature > TemperatureThreshold) ? "true" : "false");
+
+            _logger.LogInformation($"Sending message: {count}, Data: [{dataBuffer}]");
+            await s_deviceClient.SendEventAsync(eventMessage);
+        }
+
+        private async Task ReceiveMessageAndCompleteAsync()
+        {
+            using Message receivedMessage = await s_deviceClient.ReceiveAsync(s_sleepDuration).ConfigureAwait(false);
+            if (receivedMessage != null)
+            {
+                string messageData = Encoding.ASCII.GetString(receivedMessage.GetBytes());
+                _logger.LogInformation($"Received message: {messageData}");
+
+                int propCount = 0;
+                foreach (var prop in receivedMessage.Properties)
+                {
+                    _logger.LogInformation($"Property[{propCount++}> Key={prop.Key} : Value={prop.Value}");
+                }
+
+                await s_deviceClient.CompleteAsync(receivedMessage).ConfigureAwait(false);
```

Comment:
We aren't consistently using ConfigureAwait(false) everywhere.

Do we want to use it in the sample or not?

Reply:
We aren't using it in console apps, I'll remove this